src/ray/raylet/node_manager.cc

```diff
@@ -580,28 +580,26 @@ void NodeManager::ScheduleTasks() {
     if (client_id == gcs_client_->client_table().GetLocalClientId()) {
       local_task_ids.insert(task_id);
     } else {
+      // TODO(atumanov): need a better interface for task exit on forward.
       auto tasks = local_queues_.RemoveTasks({task_id});
       RAY_CHECK(1 == tasks.size());
       Task &task = tasks.front();
       // TODO(swang): Handle forward task failure.
       RAY_CHECK_OK(ForwardTask(task, client_id));
     }
-    // Notify the task dependency manager that we no longer need this task's
-    // object dependencies.
-    // NOTE(swang): For local tasks, the scheduled task's dependencies may get
-    // evicted before it can be assigned to a worker.
-    task_dependency_manager_.UnsubscribeDependencies(task_id);
   }
 
-  // Transition locally scheduled tasks to SCHEDULED and dispatch scheduled tasks.
+  // Transition locally placed tasks to waiting or ready for dispatch.
   if (local_task_ids.size() > 0) {
     std::vector<Task> tasks = local_queues_.RemoveTasks(local_task_ids);
```

Comment:
Instead of removing tasks here, leave them in the `PLACEABLE` queue and allow `TransitionPlaceableTask` to carry out the appropriate transition (see my comment about using `MoveTasks` in `TransitionPlaceableTask`).